main.tf

```diff
@@ -1,44 +1,81 @@
-resource "azurerm_bastion_host" "this" {
-  location                  = var.location
-  name                      = var.name
-  resource_group_name       = var.resource_group_name
-  copy_paste_enabled        = var.copy_paste_enabled
-  file_copy_enabled         = var.file_copy_enabled
-  ip_connect_enabled        = var.ip_connect_enabled
-  kerberos_enabled          = var.kerberos_enabled
-  scale_units               = var.scale_units
-  session_recording_enabled = var.session_recording_enabled
-  shareable_link_enabled    = var.shareable_link_enabled
-  sku                       = var.sku
-  tags                      = var.tags
-  tunneling_enabled         = var.tunneling_enabled
-  virtual_network_id        = var.virtual_network_id
-  zones                     = var.zones
-
-  dynamic "ip_configuration" {
-    for_each = var.ip_configuration != null ? [var.ip_configuration] : []
+
+resource "azapi_resource" "bastion" {
+  count = var.sku == "Developer" ? 0 : 1
 
-    content {
-      name                 = coalesce(var.ip_configuration.name, "ipconfig-${var.name}")
-      public_ip_address_id = local.public_ip_resource_id
-      subnet_id            = var.ip_configuration.subnet_id
+  type = "Microsoft.Network/bastionHosts@2024-05-01"
+  body = {
+    sku = {
+      name = var.sku
+    }
+    zones = var.zones
+    properties = {
+      disableCopyPaste         = var.copy_paste_disabled
+      enableFileCopy           = var.file_copy_enabled
+      enableIpConnect          = var.ip_connect_enabled
+      enableKerberos           = var.kerberos_enabled
+      enablePrivateOnlyBastion = var.private_only

```

Comment:
Should we add `_enabled` to this variable for consistency?

Reply:
Sure thing, done.